tests/issues/issue-1649.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+return [
+    'issue-1649-A' => [
+        'ua' => 'gvfs/1.22.2',
+        'properties' => [
+            'Comment' => 'gvfs 1.22',
+            'Browser' => 'gvfs',
+            'Browser_Type' => 'Tool',
+            'Browser_Bits' => '32',
+            'Browser_Maker' => 'The GNOME Project',
+            'Browser_Modus' => 'unknown',
+            'Version' => '1.22',
+            'MajorVer' => '1',
+            'MinorVer' => '22',
+            'Platform' => 'Linux',
+            'Platform_Version' => 'unknown',
+            'Platform_Description' => 'Linux',
+            'Platform_Bits' => '32',
+            'Platform_Maker' => 'Linux Foundation',
+            'Alpha' => false,
+            'Beta' => false,
+            'Win16' => false,
+            'Win32' => false,
+            'Win64' => false,
+            'Frames' => false,
+            'IFrames' => false,
+            'Tables' => false,
+            'Cookies' => false,
+            'BackgroundSounds' => false,
+            'JavaScript' => false,
+            'VBScript' => false,
+            'JavaApplets' => false,
+            'ActiveXControls' => false,
+            'isMobileDevice' => false,
+            'isTablet' => false,
+            'isSyndicationReader' => false,
+            'Crawler' => true,
+            'isFake' => false,
+            'isAnonymized' => false,
+            'isModified' => false,
+            'CssVersion' => '0',
+            'AolVersion' => '0',
+            'Device_Name' => 'Linux Desktop',
+            'Device_Maker' => 'unknown',
+            'Device_Type' => 'Desktop',
+            'Device_Pointing_Method' => 'mouse',
+            'Device_Code_Name' => 'Linux Desktop',
+            'Device_Brand_Name' => 'unknown',
+            'RenderingEngine_Name' => 'unknown',
+            'RenderingEngine_Version' => 'unknown',
+            'RenderingEngine_Maker' => 'unknown',
+        ],
+        'lite' => false,
+        'standard' => true,
+        'full' => true,
+    ],
+    'issue-1649-B' => [
+        'ua' => 'gvfs/1.36.2',
```

Comment:
Shouldn't this parse as 1.36?

If you're adding these to try to add coverage for the `generic` file, I generally try not to worry about those files, unless the agent has no version info in it. The issue here is that when 1.36 is released, and somebody adds the version to the file, this test will fail.

Reply:
how do you propose to cover generic ?

Reply:
You'll find that most of the generic files aren't covered for this reason.

We **could** add fake agents to the tests that the generic files would pick up, but that's a lot of maintenance, overhead, time for little value (IMO).

As I mentioned in #1670, we do have a file where we've been storing "fake" agents that do get picked up by generic files, that will increase coverage of the generic files, but that's typically when a UA is found that shouldn't be parsed as it looks (Say, Chrome on Windows 98 for example. We know that Chrome never supported Windows 98, so I've added such agents to that file to **prevent** that platform from being added to the Chrome files). If there was a version of an app/browser that was never released (and never will be), then I think that sort of agent could be added to that invalid-versions file, but I wouldn't add "might be released in the future" versions, as it's just more work for when that version is actually released and added to the files.